docs/guides/deprecating-content.rst

```diff
@@ -0,0 +1,99 @@
+Deprecating Content on Read the Docs
+====================================
+
+When you deprecate a feature from your project,
+you may want to deprecate its docs as well,
+and stop your users from reading that content.
+
+Deprecating content may sound as easy as delete it,
+but doing that will break existing links,
+and you don't necessary want to make the content inaccessible.
+Here you'll find some tips on how to use Read the Docs to deprecate your content
+progressively and in non harmful ways.
+
+Deprecating versions
+--------------------
+
+If you have multiple versions of your project,
+it makes sense to have its :doc:`documentation versioned </versions>` as well.
+For example, if you have the following versions and want to deprecate V1.
+
+- project.readthedocs.io/en/v1/
+- project.readthedocs.io/en/v2/
+- project.readthedocs.io/en/v3/
+
+For cases like this you can *hide* a version,
+hidden versions won't be listed in the versions menu of your docs,
+and they will be listed in a :ref:`robots.txt file <hosting:custom robots.txt pages>`
+to stop search engines of showing results for that version::
+
+   # project.readthedocs.io/robots.txt
+
+   User-agent: *
+
+   Disallow: /en/v1/ # Hidden version
+
+Users can still see all versions in the dashboard of your project.
+To hide a version go to your project and click on :guilabel:`Versions` > :guilabel:`Edit`,
+and mark the :guilabel:`Hidden` option. Check :ref:`versions:Version States` for more information.
+
+.. note::
+
+   If the versions of your project follow the semver_ convention,
+   you can activate the :ref:`versions:version warning` option for your project.
+   A banner with a warning and linking to the stable version
+   will be shown on all versions that are lower than the stable one.
+
+   .. _semver: https://semver.org/
+
+Deprecating content
+-------------------
+
+You may not always want to deprecate a version, but deprecate some pages.
+For example, if you have documentation about two APIs and you want to deprecate V1:
+
+- project.readthedocs.io/en/latest/api/v1.html
+- project.readthedocs.io/en/latest/api/v2.html
+
+A simple way is just adding a warning at the top of the page,
+this will warn users visiting that page,
+but it won't stop users from being redirected to that page from search results.
+You can add an entry of that page in a :ref:`custom robots.txt <hosting:custom robots.txt pages>` file
+to avoid search engines of showing those results. For example::
+
+   # robots.txt
+
+   User-agent: *
+
+   Disallow: /en/latest/api/v1.html # Deprecated API
+
+But your users will still see search results from that page if they use the search from your docs.
+With Read the Docs you can set a :ref:`custom rank per pages <config-file/v2:search.ranking>`.
+For example:
+
+.. code-block:: yaml
+
+   # .readthedocs.yaml
+
+   version: 2
+   search:
+      ranking:
+         api/v1.html: -1
```

Comment:
Same here with the `robots.txt` and `yaml` from this section. I'd say it's enough by mentioning it and pointing to the `search.ranking` config. That way, we keep this documentation in only one place. Users can just hover over the `:ref:` and immediately see how the config file looks like.

Reply:
I'm not sure, this shows how to use the feature with the current example, and it's easy to copy/paste.